src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -605,6 +609,82 @@ public DatasetDataModel retrieveData(
     }
   }
 
+  public ColumnStatisticsModel retrieveColumnStatistics(
+      AuthenticatedUserRequest userRequest,
+      UUID datasetId,
+      String tableName,
+      String columnName,
+      String filter) {
+    Dataset dataset = retrieve(datasetId);
+
+    Column column =
+        dataset
+            .getTableByName(tableName)
+            .orElseThrow(
+                () ->
+                    new InvalidTableException(
+                        "No dataset table exists with the name: " + tableName))
+            .getColumnByName(columnName)
+            .orElseThrow(
+                () ->
+                    new InvalidColumnException(
+                        "No column exists in table "
+                            + tableName
+                            + " with column name "
+                            + columnName));
+
+    var cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+
+    if (cloudPlatformWrapper.isGcp()) {
+      try {
+        String bqFormattedTableName = PDAO_PREFIX + dataset.getName() + "." + tableName;
+        if (column.isDoubleType()) {
+          return BigQueryPdao.getStatsForDoubleColumn(
```

Comment:
Can you share the rationale behind making these methods static? I've found that can make mocking / testing more difficult.

Reply:
There's just precedence in BigQueryPdao -- all of the methods in this class are static and there already a static context reference to BigQueryPdao in DatasetService. Would you suggest reworking this?

Reply:
Ah, if there's existing precedence I think it would be OOS to rework as part of this PR.